trackers/core/bytetrack/tracker.py

```diff
@@ -0,0 +1,376 @@
+from copy import deepcopy
+from typing import Optional
+
+import numpy as np
+import supervision as sv
+from scipy.optimize import linear_sum_assignment
+from scipy.spatial.distance import cdist
+
+from trackers.core.base import BaseTracker
+from trackers.core.bytetrack.kalman_box_tracker import ByteTrackKalmanBoxTracker
+from trackers.core.deepsort.feature_extractor import DeepSORTFeatureExtractor
+from trackers.utils.sort_utils import (
+    get_alive_trackers,
+    get_iou_matrix,
+)
+
+
+class ByteTrackTracker(BaseTracker):
+    """Implements ByteTrack.
+
+    ByteTrack is a simple, effective, and generic multi-object tracking method
+    that improves upon tracking-by-detection by associating *every* detection box
+    instead of discarding low-score ones. It uses a two-stage association process
+    and builds on established techniques like the Kalman filter for motion prediction
+    and the Hungarian algorithm for data association.
+
+    Args:
+        lost_track_buffer (int): Number of frames to buffer when a track is lost.
+            Increasing lost_track_buffer enhances occlusion handling, significantly
+            improving tracking through occlusions, but may increase the possibility
+            of ID switching for objects with similar appearance.
+        frame_rate (float): Frame rate of the video (frames per second).
+            Used to calculate the maximum time a track can be lost.
+        track_activation_threshold (float): Detection confidence threshold
+            for track activation. Only detections with confidence above this
+            threshold will create new tracks. Increasing this threshold
+            reduces false positives but may miss real objects with low confidence.
+        minimum_consecutive_frames (int): Number of consecutive frames that an object
+            must be tracked before it is considered a 'valid' track. Increasing
+            `minimum_consecutive_frames` prevents the creation of accidental tracks
+            from false detection or double detection, but risks missing shorter
+            tracks. Before the tracker is considered valid, it will be assigned
+            `-1` as its `tracker_id`.
+        minimum_iou_threshold (float): IOU threshold for associating detections to
+            existing tracks.
+        high_prob_boxes_threshold (float): threshold for assigning predicted boxes to high probability class.
```

Comment:
Please mention what would be the effect of increasing and decreasing this value.